diff --git a/.gitignore b/.gitignore
index ae05708..feeae4a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 !subprojects/*.wrap
 *.swp
+*.o
 .clang_complete
 .DS_Store
 .cache/
@@ -18,4 +19,5 @@ xcuserdata
 .clang-tidy
 compile_flags.txt
 .vscode
-.mesonpy*
\ No newline at end of file
+.mesonpy*
+src/proto/
diff --git a/src/chess/board.cc b/src/chess/board.cc
index 59bc0c3..ea1a280 100644
--- a/src/chess/board.cc
+++ b/src/chess/board.cc
@@ -34,7 +34,9 @@
 #include <cstring>
 #include <sstream>
 #include <utility>
+#ifndef NDEBUG
 #include <absl/cleanup/cleanup.h>
+#endif
 
 #include "utils/exception.h"
 
diff --git a/src/main.cc b/src/main.cc
index f6d90c0..625e956 100644
--- a/src/main.cc
+++ b/src/main.cc
@@ -29,12 +29,8 @@
 #include "default_search.h"
 #include "engine.h"
 #include "search/register.h"
-#include "selfplay/loop.h"
 #include "tools/backendbench.h"
 #include "tools/benchmark.h"
-#include "tools/describenet.h"
-#include "tools/leela2onnx.h"
-#include "tools/onnx2leela.h"
 #include "utils/commandline.h"
 #include "utils/esc_codes.h"
 #include "utils/logging.h"
@@ -89,16 +85,10 @@ int main(int argc, const char** argv) {
 
     CommandLine::Init(argc, argv);
     if (CommandLine::BinaryName().find("simple") == std::string::npos) {
-      CommandLine::RegisterMode("selfplay", "Play games with itself");
       CommandLine::RegisterMode("benchmark", "Quick benchmark");
       CommandLine::RegisterMode("bench", "Very quick benchmark");
       CommandLine::RegisterMode("backendbench",
                                 "Quick benchmark of backend only");
-      CommandLine::RegisterMode("leela2onnx", "Convert Leela network to ONNX.");
-      CommandLine::RegisterMode("onnx2leela",
-                                "Convert ONNX network to Leela net.");
-      CommandLine::RegisterMode("describenet",
-                                "Shows details about the Leela network.");
     }
     for (const std::string_view search_name :
          SearchManager::Get()->GetSearchNames()) {
@@ -107,12 +97,7 @@ int main(int argc, const char** argv) {
           "Use \"" + std::string(search_name) + "\" search");
     }
 
-    if (CommandLine::ConsumeCommand("selfplay")) {
-      // Selfplay mode.
-      StdoutUciResponder uci_responder;
-      SelfPlayLoop loop(&uci_responder);
-      loop.Run();
-    } else if (CommandLine::ConsumeCommand("benchmark")) {
+    if (CommandLine::ConsumeCommand("benchmark")) {
       // Benchmark mode, longer version.
       Benchmark benchmark;
       benchmark.Run();
@@ -124,12 +109,6 @@ int main(int argc, const char** argv) {
       // Backend Benchmark mode.
       BackendBenchmark benchmark;
       benchmark.Run();
-    } else if (CommandLine::ConsumeCommand("leela2onnx")) {
-      lczero::ConvertLeelaToOnnx();
-    } else if (CommandLine::ConsumeCommand("onnx2leela")) {
-      lczero::ConvertOnnxToLeela();
-    } else if (CommandLine::ConsumeCommand("describenet")) {
-      lczero::DescribeNetworkCmd();
     } else {
       lczero::ChooseAndRunEngine();
     }
diff --git a/src/default_backend.h b/src/default_backend.h
new file mode 100644
index 0000000..e69de29
diff --git a/src/default_search.h b/src/default_search.h
new file mode 100644
index 0000000..e69de29
